crates/ironrdp-rdpdr/src/pdu/esc.rs

```diff
@@ -0,0 +1,515 @@
+            0x0009002C => Ok(ScardIoctlCode::ListReadersW),
+            0x00090050 => Ok(ScardIoctlCode::IntroduceReaderGroupA),
+            0x00090054 => Ok(ScardIoctlCode::IntroduceReaderGroupW),
+            0x00090058 => Ok(ScardIoctlCode::ForgetReaderGroupA),
+            0x0009005C => Ok(ScardIoctlCode::ForgetReaderGroupW),
+            0x00090060 => Ok(ScardIoctlCode::IntroduceReaderA),
+            0x00090064 => Ok(ScardIoctlCode::IntroduceReaderW),
+            0x00090068 => Ok(ScardIoctlCode::ForgetReaderA),
+            0x0009006C => Ok(ScardIoctlCode::ForgetReaderW),
+            0x00090070 => Ok(ScardIoctlCode::AddReaderToGroupA),
+            0x00090074 => Ok(ScardIoctlCode::AddReaderToGroupW),
+            0x00090078 => Ok(ScardIoctlCode::RemoveReaderFromGroupA),
+            0x0009007C => Ok(ScardIoctlCode::RemoveReaderFromGroupW),
+            0x00090098 => Ok(ScardIoctlCode::LocateCardsA),
+            0x0009009C => Ok(ScardIoctlCode::LocateCardsW),
+            0x000900A0 => Ok(ScardIoctlCode::GetStatusChangeA),
+            0x000900A4 => Ok(ScardIoctlCode::GetStatusChangeW),
+            0x000900A8 => Ok(ScardIoctlCode::Cancel),
+            0x000900AC => Ok(ScardIoctlCode::ConnectA),
+            0x000900B0 => Ok(ScardIoctlCode::ConnectW),
+            0x000900B4 => Ok(ScardIoctlCode::Reconnect),
+            0x000900B8 => Ok(ScardIoctlCode::Disconnect),
+            0x000900BC => Ok(ScardIoctlCode::BeginTransaction),
+            0x000900C0 => Ok(ScardIoctlCode::EndTransaction),
+            0x000900C4 => Ok(ScardIoctlCode::State),
+            0x000900C8 => Ok(ScardIoctlCode::StatusA),
+            0x000900CC => Ok(ScardIoctlCode::StatusW),
+            0x000900D0 => Ok(ScardIoctlCode::Transmit),
+            0x000900D4 => Ok(ScardIoctlCode::Control),
+            0x000900D8 => Ok(ScardIoctlCode::GetAttrib),
+            0x000900DC => Ok(ScardIoctlCode::SetAttrib),
+            0x000900E0 => Ok(ScardIoctlCode::AccessStartedEvent),
+            0x000900E4 => Ok(ScardIoctlCode::ReleaseTartedEvent),
+            0x000900E8 => Ok(ScardIoctlCode::LocateCardsByAtrA),
+            0x000900EC => Ok(ScardIoctlCode::LocateCardsByAtrW),
+            0x000900F0 => Ok(ScardIoctlCode::ReadCacheA),
+            0x000900F4 => Ok(ScardIoctlCode::ReadCacheW),
+            0x000900F8 => Ok(ScardIoctlCode::WriteCacheA),
+            0x000900FC => Ok(ScardIoctlCode::WriteCacheW),
+            0x00090100 => Ok(ScardIoctlCode::GetTransmitCount),
+            0x00090104 => Ok(ScardIoctlCode::GetReaderIcon),
+            0x00090108 => Ok(ScardIoctlCode::GetDeviceTypeId),
+            _ => {
+                error!("Unsupported ScardIoctlCode: 0x{:08x}", value);
+                Err(invalid_message_err!("try_from", "ScardIoctlCode", "unsupported value"))
+            }
+        }
+    }
+}
+
+/// Allow [`ScardIoctlCode`] to be used as an [`IoctlCode`].
+impl IoctlCode for ScardIoctlCode {}
```

Comment:
Nit/Question: should this be named this way?
```suggestion
impl IoCtlCode for ScardIoCtlCode {}
```

Reply:
Sure, I was just copying what we had in Teleport but that works for me: 5376e0b5aa1421b73bcc899b7df69adfc79db51f